src/MPoly.jl

```diff
@@ -445,8 +451,17 @@ function content(a::MPolyRingElem{T}) where T <: RingElement
    return z
 end
 
+
+function is_nilpotent(f::T) where {T<:MPolyRingElem}
+#  is_domain_type(elem_type(coefficient_ring(f))) && return is_zero(f)
```

Comment:
That's not right. You need
```suggestion
function is_nilpotent(f::MPolyRingElem{T}) where {T <: RingElement}
```
`is_domain_type` needs to be given the element type of the ring we want to consider -- and that is also precisely the type parameter used for `MPolyRingElem`, `PolyRingElem`, `MatElem` etc. etc.

Reply:
Ah, but thinking more about this, of course we also implement `is_domain_type` "recursively" for `MPolyRingElem`, so this will work after all. Sorry for the noise